src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -496,14 +504,65 @@ public TStructuralType Patch(TStructuralType original)
         /// <summary>
         /// Overwrites the <paramref name="original"/> entity with the values stored in this delta.
         /// </summary>
+        /// <param name="original">The entity to be updated.</param>
+        /// <param name="apiHandler">API Handler for the entity.</param>
+        public void Patch(TStructuralType original, IODataAPIHandler apiHandler)
+        {
+            Patch(original, apiHandler, null);
+        }
+
+        /// <summary>
+        /// Overwrites the <paramref name="original"/> entity with the values stored in this delta.
+        /// </summary>
+        /// <param name="original">The entity to be updated.</param>
+        /// <param name="apiHandler">API Handler for the entity.</param>
+        /// <param name="apiHandlerFactory">API Handler Factory.</param>
+        public void Patch(TStructuralType original, IODataAPIHandler apiHandler, ODataAPIHandlerFactory apiHandlerFactory)
```

Comment:
What did you say happens when someone passes non-null values for both the `apiHandler` and `apiHandlerFactory`? Which gets used where?

Reply:
`ApiHandlerFactory.GetHandler(ODataPath)` is used where we have `@odata.id`.
`Handler.GetNestedHandler(...)` is used to get handler for nested resources.